pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -85,6 +97,146 @@ func fixVirtualBoxInstallation() (bool, error) {
 	return true, nil
 }
 
+func tryRemoveDestFile(url string, destDir string) error {
+	u, err := neturl.Parse(url)
+	if err != nil {
+		return fmt.Errorf("Cannot parse URL %s", url)
+	}
+
+	urlPath, err := neturl.PathUnescape(u.EscapedPath())
+	if err != nil {
+		return fmt.Errorf("Cannot unescape URL path %s", urlPath)
+	}
+	destFilename := path.Base(urlPath)
+	destPath := filepath.Join(destDir, destFilename)
+	err = os.Remove(destPath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Could not remove %s: %v", destPath, err)
+	}
+
+	return nil
+}
+
+func download(url string, destDir string, mode os.FileMode) (string, error) {
+	err := os.MkdirAll(destDir, 0111 | mode)
+	if err != nil && !os.IsExist(err) {
+		return "", fmt.Errorf("Cannot create directory %s", destDir)
+	}
+
+	// If the destination file already exists, dl.Download may not be able to
+	// overwrite it if we made it suid. We can however delete it beforehand.
+	err = tryRemoveDestFile(url, destDir)
+	if err != nil {
+		return "", err
+	}
+
+	filename, err := dl.Download(url, destDir)
+	if err != nil {
+		return "", err
+	}
+
+	err = os.Chmod(filename, mode)
+	if err != nil {
+		os.Remove(filename)
+		return "", err
+	}
+
+	return filename, nil
+}
+
+func setSuid(path string) (error) {
+	logging.DebugF("Making %s suid", path)
+
+	stdOut, stdErr, err := crcos.RunWithPrivilege("chown", "root:wheel", path)
+	if err != nil {
+		return fmt.Errorf("Unable to set ownership of %s to root:wheel: %s %v: %s",
+			path, stdOut, err, stdErr)
+	}
+
+	/* Can't do this before the chown as the chown will reset the suid bit */
+	stdOut, stdErr, err = crcos.RunWithPrivilege("chmod", "u+s", path)
+	if err != nil {
+		return fmt.Errorf("Unable to set suid bit on %s: %s %v: %s", path, stdOut, err, stdErr)
+	}
+	return nil
+}
+
+func checkSuid(path string) (bool, error) {
+	fi, err := os.Stat(path)
+	if (err != nil) {
+		return false, err
+	}
+	if fi.Mode()&os.ModeSetuid == 0 {
+		return false, fmt.Errorf("%s does not have the SUID bit set (%s)", path, fi.Mode().String())
+	}
+	if fi.Sys().(*syscall.Stat_t).Uid != 0 {
+		return false, fmt.Errorf("%s is not owned by root", path)
+	}
+
+	return true, nil
+}
+
+func checkHyperKitInstalled() (bool, error) {
+	logging.DebugF("Checking if hyperkit is installed")
+	hyperkitPath := filepath.Join(hyperkitCacheDir, "hyperkit")
+	err := unix.Access(hyperkitPath, unix.X_OK)
+	if err != nil {
+		logging.DebugF("%s not executable", hyperkitPath)
+		return false, err
+	}
+
+	return checkSuid(hyperkitPath)
+}
+
+func fixHyperKitInstallation() (bool, error) {
+	hyperkitFile, err := download(hyperkitDownloadURL, hyperkitCacheDir, 0755)
```

Comment:
shouldn't download happen to a temporary directory, verified against a hash, and then confirm it was downloaded correctly.

Reply:
Yes most likely, though we currently don't do that for any binary that we download (virtualbox.dmg, oc, ...). Or maybe we could ship all of this in a .dmg?

Reply:
I filed https://github.com/code-ready/crc/issues/321 for that issue, I'd suggest not blocking the merging of this PR on this.

Reply:
we had already agreed on this on slack